src/dotnet/commands/dotnet-migrate/MigrateCommand.cs

```diff
@@ -132,7 +133,15 @@ private void UpdateSolutionFile(MigrationReport migrationReport)
                 }
                 else
                 {
-                    csprojFilesToAdd.Add(Path.Combine(report.ProjectDirectory, migratedProjectName));
+                    var csprojPath = Path.Combine(relReportPath, migratedProjectName);
+                    var slnAlreadyContainsMigratedCsproj = _slnFile.Projects
+                        .Where(p => p.FilePath == csprojPath)
+                        .Any();
+
+                    if (!slnAlreadyContainsMigratedCsproj)
```

Comment:
The HashSet guards against adding two (which I think is a good change). But this code protects from adding one. In the repro, we shouldn't even be trying to add one (so the HashSet doesn't guard against that).